### DISCUSSION TITLE
webgl state diagram : draw on cube

### DISCUSSION THREAD

**Discussion by letochagone** [gfxfundamentals/webgl-fundamentals#420] [Q&A] [answered]
In the webgl-state-diagram I wonder how the "draw cube on cube" example can work.
That is, one of the textures uses
 

> **gl.LUMINANCE, // internal format**, 

but there is no : 

> **gl.pixelStorei(gl.UNPACK_ALIGNMENT, 1);**

I don't think this is a mistake, but rather a misunderstanding on my part.
However I couldn't find where to ask this question other than in "issues".

**Comment by greggman** [comment, marked as answer]
```
gl.pixelStorei(gl.UNPACK_ALIGNMENT, 1);
```

is required if a row of the data is not a multiple of 4 bytes (the default UNPACK_ALIGNMENT setting). In the example, 16 bytes are being passed in. 4 rows of 4 bytes each. Because each row is 4 bytes and 4 is a multiple of 4 it works as is.
